bindings/c/src/lib.rs

```diff
@@ -0,0 +1,305 @@
+use ffi_interface::{
+    deserialize_proof_query, deserialize_proof_query_uncompressed, deserialize_verifier_query,
+    deserialize_verifier_query_uncompressed, fr_from_le_bytes, get_tree_key_hash, Context,
+};
+use ipa_multipoint::committer::Committer;
+use ipa_multipoint::multiproof::{MultiPoint, MultiPointProof, ProverQuery, VerifierQuery};
+use ipa_multipoint::transcript::Transcript;
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+pub extern "C" fn context_new() -> *mut Context {
+    let ctx = Box::new(Context::default());
+    Box::into_raw(ctx)
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+pub extern "C" fn context_free(ctx: *mut Context) {
+    if ctx.is_null() {
+        return;
+    }
+    unsafe {
+        let _ = Box::from_raw(ctx);
+    }
+}
+
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+#[no_mangle]
+pub extern "C" fn pedersen_hash(
+    ctx: *mut Context,
+    address: *const u8,
+    tree_index_le: *const u8,
+    out: *mut u8,
+) {
+    if ctx.is_null() || address.is_null() || tree_index_le.is_null() || out.is_null() {
+        // TODO: We have ommited the error handling for null pointers at the moment.
+        // TODO: Likely will panic in this case.
```

Comment:
We can still merge with this TODO, just noting that we probably won't panic in the future since we ideally don't want the library to panic

Reply:
i just copied this direct from the kzg project, wanted to keep things and layout similar to the kzg library